test/pkg.jl

```diff
@@ -100,6 +100,8 @@ temp_pkg_dir() do
     str = chomp(takebuf_string(iob))
     @test endswith(str, string(Pkg.installed("Example")))
     @test isempty(Pkg.dependents("Example"))
+    # Checking to see dependency reading
+    @test Pkg.dependencies("Example") == Dict("julia" => Pkg.Types.VersionSet(v"0.3.0", typemax(VersionNumber)))
```

Comment:
That sounds like it would be difficult to predict the behavior if what version it returns information about depends on what's installed at the time. Maybe if `version` is given as `installed` then behave this way but error if not installed?